### PR TITLE
completed week 2 assessment

### PR DESCRIPTION
**Pull request by segjr**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅] Coding Challenges
- [✅] Coding Concepts
- [✅] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:
I really appreciate the previous feedback for week one. I'd appreciate any feedback on current level of understand and areas of improvement. Thank you.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -35,10 +35,36 @@ const num2 = 0
...
+// Have the function return the string interpolation `${number} is not divisible by three
+// console log new function with challenge parameters
+
+let byThree = (number) => {
+    if(number % 3 == 0) {
+        return `${number} is divisible by three`
+    }else if(number % 3 !== 0) {
+        return `${number} is not divisible by three`
+    }
+}
```
Comment (elysemontano) [inline code-challenges.test.js:63]: This looks great!  If you want to stretch yourself, possibly consider edge cases in your code and tests.

File: code-challenges.test.js [hunk 2]
```diff
@@ -49,10 +75,34 @@ const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
...
 
+// Pseudo Code:
+// This program will allow the user to input an array of strings and return the array with the strings capitalized.
+// Create a function expression. (upCase)
+// Utilize a method on the parameter to return a new array (.map())
+    // Within .map, utilize a method on the value parameter that will return a string object's index value. (charAt())
+    // Within .map, utilize a method on the aforementioned method that will convert the casing of a string object to uppercase letters. (toUpperCase())
+    // Within .map, utilize a method on a separated copy of the value parameter that will return the substring of a string by index value parameter. (.substr())
+    // Within .map, utilize an operator to merge and return the two value parameters for the .map value. (+)
+// Console log the new function with the challenge parameters.
```
Comment (elysemontano) [inline code-challenges.test.js:95]: Great pseudocode!!

File: code-challenges.test.js [hunk 3]
```diff
@@ -65,5 +115,37 @@ const vowelTester2 = "academy"
...
+let theVowel = (string) => {
+    let v = ["a","e","i","o","u"]
+    for(i = 0; i < string.length; i++) {
+        for(j = 0; j < v.length; j++) {
+            if(string[i] == v[j]) {
+            return i
+            }
+        }
+    }
+}
```
Comment (elysemontano) [inline code-challenges.test.js:147]: Ooo, nested for loops!  Many people get very confused with nested for loops, but this looks great!

File: coding-concepts.js [hunk 4]
```diff
@@ -67,7 +67,7 @@ class Learn {
   }
 }
 var learnStudent = new Learn("George")
-// console.log(learnStudent)
+console.log(learnStudent)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: Learn {student: "George", cohort: "Bravo", year: 2022}
+// b) Verify and explain: My initial answer is correct because the constructor method takes the given parameter and uses it as the name within the method when it returns the object created from the key:value pairs. The above method uses [this.] to reference the keys within the Learn class. The constructor method is used to create a new object. Since it is within the Learn class, Learn will be returned to associate the object with a class. Learn will work as a function to produce an object with the values assigned from the constructor keys.
```
Comment (elysemontano) [inline coding-concepts.js:73]: Great job on this section!

File: interview-questions.md [hunk 5]
```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: Hoisting refers to the process of declarations being moved to the top of the current scope before code is executed. Coincidentally, I discovered this definition when trying to discover important concepts of javascript, prior to LEARN Academy. 
 
-  Researched answer:
+  Researched answer: Hoisting is a default behavior in javascript. A benefit is that it allows functions and variables to be called before they are scripted.
```
Comment (elysemontano) [inline interview-questions.md:53]: Excellent answers for this entire section!

File: code-challenges.test.js [hunk 6]
```diff
@@ -35,10 +35,36 @@ const num2 = 0
 const num3 = -7
 // Expected output: "-7 is not divisible by three"
 
+describe("byThree", () => {
+    it("takes in a number as an argument and returns if the number is divisible by three or not", () => {
+        expect(byThree(num1)).toEqual("15 is divisible by three")
+        expect(byThree(num2)).toEqual("0 is divisible by three")
+        expect(byThree(num3)).toEqual("-7 is not divisible by three")
+    })
+})
```
Comment (elysemontano) [inline code-challenges.test.js:44]: I often recommend placing the test variables being used inside the it block just before the expect statement so they are confined to the scope they are used in.

**Review by elysemontano — APPROVED** [review]
Crushed this assessment Samuel!!  This assessment shows a Competent level of understanding, providing excellent pseudocode, well written tests, superb logic and code.  Your answers in the interview and code concept sections were really well thought out and nicely worded.  Left some comments for possible things to consider for future assessments.  Keep up the great work!
